woocommerce/woocommerce-android
# Update/build lane

This PR adds GitHub releases creation to the lanes that build the beta and the final artefacts. 
It also tweaks the `code_freeze` lane to update the `release_notes.txt` file with the draft release notes so that they are available when building the first beta.  

## To Test
_1. Code freeze lane_
You can simulate a complete code freeze, but since the work the new action does is unrelated to the other steps, it can be safely tested alone. 

1.a Checkout a new release/<something> branch from this one.
1.b Comment out all the steps in `code_freeze` other than line 53 and 54. Commit the changes.
1.c `bundle install`
1.c Run `bundle exec fastlane code_freeze` and verify that `WooCommerce/metadata/release_notes.txt` has been updated with the content of the main `RELEASE_NOTES.txt` related to the current version.
1.d Delete the test branch

_2. Beta build_
2.a Checkout a new release/<something> branch from this one.
2.b Edit `build.gradle` with a new beta version and versionCode. Commit the changes.
2.c Run `bundle install`.
2.d Run `bundle exec fastlane build_and_upload_beta` and verify that the a release is drafted on GitHub and that the release notes are added, the "Pre-release" option is checked and the .apk is uploaded.
2.e Delete the artifact from Google Play.
2.f Delete the test branch.

_3. Release build_
2.a Checkout a new release/<something> branch from this one.
2.b Edit `build.gradle` with a final version and versionCode. Commit the changes.
2.c Run `bundle install`.
2.d Run `bundle exec fastlane build_and_upload_release` and verify that the a release is drafted on GitHub and that the release notes are added and the .apk and .aab are uploaded.
2.e Delete the artifact from Google Play.
2.f Delete the test branch.

**Note 1**: When this is ready to be merged, a new version of `release-toolkit` should be released and the references in this PR updated. 

**Note 2**: I'm targeting the release branch because this stuff is used only on releases and I think it's better to test it sooner than later :-) If the review won't be completed before the `4.5` release, I'll retarget it. 

**Note 3**: Related `release-tookit` PR: https://github.com/wordpress-mobile/release-toolkit/pull/136

 
Update release notes:
